packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -134,9 +150,13 @@ export class DetachedFieldIndex {
 		return key;
 	}
 
-	public deleteEntry(nodeId: Delta.DetachedNodeId): void {
-		const found = deleteFromNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor);
-		assert(found, 0x7ab /* Unable to delete unknown entry */);
+	public deleteEntry(nodeId: Delta.DetachedNodeId, count: number = 1): void {
+		if (this.detachedNodeRangeMap.has(nodeId.major)) {
```

Comment:
This function used to assert when no data was found for the given major. We want to keep doing that.
You can start by calling `this.detachedNodeRangeMap.get(nodeId.major)` and assert that the return value is not undefined. That'll also get rid of the need for the no-non-null-assertion directive.